From 7738a45499e658cd004ae45a58c2c3e39e3ef812 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 26 Nov 2010 00:08:21 +0000
Subject: [PATCH] daemon: Don't pass NULL strings to g_variant_new()

---
 gnome-settings-daemon/main.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gnome-settings-daemon/main.c b/gnome-settings-daemon/main.c
index 2391ccf..780f5ca 100644
--- a/gnome-settings-daemon/main.c
+++ b/gnome-settings-daemon/main.c
@@ -114,7 +114,7 @@ got_session_proxy (GObject      *source_object,
                 startup_id = g_getenv ("DESKTOP_AUTOSTART_ID");
                 g_dbus_proxy_call (proxy,
                                    "RegisterClient",
-                                   g_variant_new ("(ss)", "gnome-settings-daemon", startup_id),
+                                   g_variant_new ("(ss)", "gnome-settings-daemon", startup_id ? startup_id : ""),
                                    G_DBUS_CALL_FLAGS_NONE,
                                    -1,
                                    NULL,
-- 
1.7.3.2

