diff -up gnome-settings-daemon-2.91.2-20101102/plugins/xsettings/gsd-xsettings-manager.c.change gnome-settings-daemon-2.91.2-20101102/plugins/xsettings/gsd-xsettings-manager.c
--- gnome-settings-daemon-2.91.2-20101102/plugins/xsettings/gsd-xsettings-manager.c.change	2010-11-02 20:16:55.247179988 -0400
+++ gnome-settings-daemon-2.91.2-20101102/plugins/xsettings/gsd-xsettings-manager.c	2010-11-02 20:17:09.784180223 -0400
@@ -701,7 +701,7 @@ gnome_xsettings_manager_start (GnomeXSet
         }
 
         list = g_hash_table_get_values (manager->priv->settings);
-        for (l = list; list != NULL; list = list->next) {
+        for (l = list; l != NULL; l = l->next) {
                 g_signal_connect (G_OBJECT (l->data), "changed",
                                   G_CALLBACK (xsettings_callback), manager);
         }
