diff -up gnome-settings-daemon-2.91.90/configure.ac.undefined gnome-settings-daemon-2.91.90/configure.ac
--- gnome-settings-daemon-2.91.90/configure.ac.undefined	2011-02-25 22:33:12.551043059 -0500
+++ gnome-settings-daemon-2.91.90/configure.ac	2011-02-25 22:34:05.670378974 -0500
@@ -281,7 +281,7 @@ AC_ARG_ENABLE(packagekit,
 
 if test x$WANT_PACKAGEKIT = xyes ; then
        PK_REQUIRED_VERSION=0.6.4
-       PKG_CHECK_MODULES(PACKAGEKIT, glib-2.0 packagekit-glib2 >= $PK_REQUIRED_VERSION upower-glib >= 0.9.1,
+       PKG_CHECK_MODULES(PACKAGEKIT, glib-2.0 packagekit-glib2 >= $PK_REQUIRED_VERSION upower-glib >= 0.9.1 gudev-1.0 libnotify >= $LIBNOTIFY_REQUIRED_VERSION,
              [have_packagekit=true
               AC_DEFINE(HAVE_PACKAGEKIT, 1, [Define if PackageKit should be used])],
              [have_packagekit=false])
