diff -up gnome-settings-daemon-2.25.90/plugins/mouse/gsd-mouse-manager.c.catch-deviceadded gnome-settings-daemon-2.25.90/plugins/mouse/gsd-mouse-manager.c
--- gnome-settings-daemon-2.25.90/plugins/mouse/gsd-mouse-manager.c.catch-deviceadded	2009-02-04 23:23:55.000000000 -0500
+++ gnome-settings-daemon-2.25.90/plugins/mouse/gsd-mouse-manager.c	2009-02-04 23:22:13.000000000 -0500
@@ -320,7 +320,7 @@ devicepresence_filter (GdkXEvent *xevent
         if (xev->type == xi_presence)
         {
             XDevicePresenceNotifyEvent *dpn = (XDevicePresenceNotifyEvent *) xev;
-            if (dpn->devchange == DeviceEnabled)
+            if (dpn->devchange == DeviceEnabled || dpn->devchange == DeviceAdded)
                 set_mouse_settings ((GsdMouseManager *) data);
         }
         return GDK_FILTER_CONTINUE;
