--- gnome-settings-daemon-3.8.0/plugins/keyboard/gsd-keyboard-manager.c~	2013-03-21 17:12:16.000000000 +0900
+++ gnome-settings-daemon-3.8.0/plugins/keyboard/gsd-keyboard-manager.c	2013-04-04 15:15:46.026364282 +0900
@@ -453,7 +453,7 @@
                 { "bn_IN", "m17n:bn:inscript" },
                 { "gu_IN", "m17n:gu:inscript" },
                 { "hi_IN", "m17n:hi:inscript" },
-                { "ja_JP", "anthy" },
+                { "ja_JP", "kkc" },
                 { "kn_IN", "m17n:kn:kgp" },
                 { "ko_KR", "hangul" },
                 { "mai_IN", "m17n:mai:inscript" },
@@ -464,7 +464,7 @@
                 { "sd_IN", "m17n:sd:inscript" },
                 { "ta_IN", "m17n:ta:tamil99" },
                 { "te_IN", "m17n:te:inscript" },
-                { "zh_CN", "pinyin" },
+                { "zh_CN", "libpinyin" },
                 { "zh_HK", "cangjie3" },
                 { "zh_TW", "chewing" },
         };
