diff -up gnome-settings-daemon-2.31.3/plugins/xrandr/gsd-xrandr-manager.c.display-panel gnome-settings-daemon-2.31.3/plugins/xrandr/gsd-xrandr-manager.c
--- gnome-settings-daemon-2.31.3/plugins/xrandr/gsd-xrandr-manager.c.display-panel	2010-06-16 08:12:53.017980209 -0400
+++ gnome-settings-daemon-2.31.3/plugins/xrandr/gsd-xrandr-manager.c	2010-06-16 08:13:45.060956325 -0400
@@ -76,7 +76,7 @@
 #define GSD_XRANDR_ICON_NAME "gsd-xrandr"
 
 /* executable of the control center's display configuration capplet */
-#define GSD_XRANDR_DISPLAY_CAPPLET "gnome-display-properties"
+#define GSD_XRANDR_DISPLAY_CAPPLET "gnome-control-center display"
 
 #define GSD_DBUS_PATH "/org/gnome/SettingsDaemon"
 #define GSD_DBUS_NAME "org.gnome.SettingsDaemon"
