diff -up gnome-settings-daemon-2.23.5/plugins/media-keys/actions/acme.glade.icon-names gnome-settings-daemon-2.23.5/plugins/media-keys/actions/acme.glade
--- gnome-settings-daemon-2.23.5/plugins/media-keys/actions/acme.glade.icon-names	2008-07-25 13:41:10.000000000 -0400
+++ gnome-settings-daemon-2.23.5/plugins/media-keys/actions/acme.glade	2008-07-25 13:41:35.000000000 -0400
@@ -20,7 +20,7 @@
 	    <widget class="GtkImage" id="acme_image">
 	      <property name="visible">True</property>
 	      <property name="icon_size">6</property>
-	      <property name="icon_name">stock_volume-max</property>
+	      <property name="icon_name">audio-volume-high</property>
 	      <property name="xalign">0.5</property>
 	      <property name="yalign">0.5</property>
 	      <property name="xpad">0</property>
