Index: gnome-settings-daemon-2.22.0/plugins/mouse/gsd-mouse-manager.c
===================================================================
--- gnome-settings-daemon-2.22.0/plugins/mouse/gsd-mouse-manager.c	(revision 224)
+++ gnome-settings-daemon-2.22.0/plugins/mouse/gsd-mouse-manager.c	(working copy)
@@ -439,8 +439,6 @@
                         XUngrabKeyboard (gdk_x11_get_default_xdisplay (),
                                          xev->xkey.time);
                 }
-
-                return GDK_FILTER_REMOVE;
         }
         return GDK_FILTER_CONTINUE;
 }
