diff -up gnome-settings-daemon-2.25.90/plugins/a11y-keyboard/gsd-a11y-keyboard-manager.c.notify-statusicon gnome-settings-daemon-2.25.90/plugins/a11y-keyboard/gsd-a11y-keyboard-manager.c
--- gnome-settings-daemon-2.25.90/plugins/a11y-keyboard/gsd-a11y-keyboard-manager.c.notify-statusicon	2009-02-05 00:40:42.000000000 -0500
+++ gnome-settings-daemon-2.25.90/plugins/a11y-keyboard/gsd-a11y-keyboard-manager.c	2009-02-05 00:48:26.000000000 -0500
@@ -544,10 +544,11 @@ ax_slowkeys_warning_post_bubble (GsdA11y
         }
 
         gsd_a11y_keyboard_manager_ensure_status_icon (manager);
-        manager->priv->notification = notify_notification_new_with_status_icon (title,
-                                                                                message,
-                                                                                "preferences-desktop-accessibility",
-                                                                                manager->priv->status_icon);
+        manager->priv->notification = notify_notification_new (title,
+                                                               message,
+                                                               "preferences-desktop-accessibility",
+                                                               NULL);
+        notify_notification_attach_to_status_icon (manager->priv->notification, manager->priv->status_icon);
         notify_notification_set_timeout (manager->priv->notification, NOTIFICATION_TIMEOUT * 1000);
 
         notify_notification_add_action (manager->priv->notification,
@@ -688,10 +689,11 @@ ax_stickykeys_warning_post_bubble (GsdA1
         }
 
         gsd_a11y_keyboard_manager_ensure_status_icon (manager);
-        manager->priv->notification = notify_notification_new_with_status_icon (title,
-                                                                                message,
-                                                                                "preferences-desktop-accessibility",
-                                                                                manager->priv->status_icon);
+        manager->priv->notification = notify_notification_new (title,
+                                                               message,
+                                                               "preferences-desktop-accessibility",
+                                                               NULL):
+        notify_notification_attach_to_status_icon (manager->priv->notification, manager->priv->status_icon);
         notify_notification_set_timeout (manager->priv->notification, NOTIFICATION_TIMEOUT * 1000);
 
         notify_notification_add_action (manager->priv->notification,
