diff -up gnome-settings-daemon-2.27.3/plugins/keyboard/gsd-keyboard-xkb.c.xklavier4 gnome-settings-daemon-2.27.3/plugins/keyboard/gsd-keyboard-xkb.c
--- gnome-settings-daemon-2.27.3/plugins/keyboard/gsd-keyboard-xkb.c.xklavier4	2009-07-01 09:19:26.936571862 -0400
+++ gnome-settings-daemon-2.27.3/plugins/keyboard/gsd-keyboard-xkb.c	2009-07-01 09:32:19.620817134 -0400
@@ -168,7 +168,7 @@ filter_xkb_config (void)
 	xkl_debug (100, "Filtering configuration against the registry\n");
 	if (!xkl_registry) {
 		xkl_registry = xkl_config_registry_get_instance (xkl_engine);
-		if (!xkl_config_registry_load (xkl_registry)) {
+		if (!xkl_config_registry_load (xkl_registry, FALSE)) {
 			g_object_unref (xkl_registry);
 			xkl_registry = NULL;
 			return FALSE;
